### PR TITLE
Correct getServletPath for a default mapping case

### PR DESCRIPTION
**Pull request by pmd1nh**

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build  #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS

**Comment by pmd1nh** [comment]
https://github.com/OpenLiberty/open-liberty/issues/4666

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -164,4 +164,7 @@ useSemiColonAsDelimiterInURI.name=Use semicolon as delimiter in URI
...
 
 servletDestroyWaitTime.name=Wait for an active request to complete
-servletDestroyWaitTime.desc=Wait time in seconds for an active request to complete when the owning application is stopped. The default value is 60 seconds. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.servletDestroyWaitTime.
\ No newline at end of file
+servletDestroyWaitTime.desc=Wait time in seconds for an active request to complete when the owning application is stopped. The default value is 60 seconds. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.servletDestroyWaitTime.
+
+servletPathForDefaultMapping.name=Set the servlet path value for a default mapping
+servletPathForDefaultMapping.desc=Set the servlet path value to the request URI minus the context path and the path info is null when a servlet is used as a default mapping. The default value is true for the Servlet 4.0 feature or above, and false for other Servlet features. Also when mapping to the /* pattern, the servlet path will be empty and the path info will start with a leading /.
```
Comment (chirp1) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: Change to: Set the servlet path value to the request URI minus the context path. The path information is null when a servlet is used as a default mapping. The default value is true for version 4.0 or later of the servlet feature. It is false for other servlet features. When mapping is to the /* pattern, the servlet path is empty and the path information starts with a leading slash (/).

Comment (chirp1) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: Set the servlet path value to the request URI minus the context path and the path info is null when a servlet is used as a default mapping. The default value is true for the Servlet 4.0 feature or above, and false for other Servlet features. Also when mapping to the /* pattern, the servlet path will be empty and the path info will start with a leading /.

**Comment by pnicolucci** [comment]
@pmd1nh I believe you'll need to update the com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.servlet-servletSpi1.0.feature file with the following:

files=dev/spi/ibm/javadoc/com.ibm.websphere.appserver.spi.servlet_2.2-javadoc.zip

The above needs to be changed to:

-files=dev/spi/ibm/javadoc/com.ibm.websphere.appserver.spi.servlet_2.3-javadoc.zip

Since the bundle version needed to be changed to 2.3 in: dev/com.ibm.websphere.appserver.spi.servlet/bnd.bnd

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS

**Comment by pmd1nh** [comment]
Updated 
the com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.servlet-servletSpi1.0.feature

Rerun build with all buckets.

**Comment by pmd1nh** [comment]
rebased with origin/integration.

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS

**Comment by pnicolucci** [comment]
A manual build that Phu kicked off to pick up the feature checker changes required in integration: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_XuHBUKYJEeiCc5H4X7yQuA&tab=com.ibm.team.build.web.ui.internal.editors.result.overview.OverviewPage

**Comment by pmd1nh** [comment]
rebased on integration.

#build

**Pull request merged**